test/demo/utils/hydrateNotification.ts

```diff
@@ -0,0 +1,31 @@
+import {NotificationOptions, Notification} from '../../../src/client';
+
+export function assertHydratedCorrectly(options: NotificationOptions, fullNote: Notification): void {
+    // Handle `id` and `date` seperately as these cannot be predicted if undefined
+    if (options.id) {
+        expect(fullNote.id).toEqual(options.id);
+    }
+    if (options.date) {
+        expect(fullNote.date).toEqual(options.date);
+    }
+
+    // customData is never changed in any way by the provider
+    expect(fullNote.customData).toEqual(options.customData);
+
+    // Hydrate the buttons array if specified
+    if (Array.isArray(options.buttons)) {
```

Comment:
Can this just be `if (options.buttons)` ?

Reply:
I wanted to mirror the check we make in the provider, which includes the `isArray` check